### ISSUE TITLE
Iterm2 image preview image_decoder crash

### ISSUE THREAD

**Issue by Bugxyb** [ranger/ranger#1868]
#### Runtime Environment

- Operating system and version: macOS 10.15.3
- Terminal emulator and version: iterm2 3.3.9
- Python version: 2.7.6
- Ranger version/commit: 1.9.3
- Locale: None
- image_preview_method iterm2

#### Current Behavior
When previewing Huge images(like 2560*1918), the iterm2 will dump: 
![image](https://user-images.githubusercontent.com/6049660/75845561-72680d00-5e14-11ea-82a2-66de332d6455.png)
the smaller images is OK. Maybe it is memory crash cause by b64 content over size. and I read the source code, find implement the imgcat. 

#### Expected Behavior
1. update the imgcat, support the image resize. image preview can show the thumbnail. huge image affects typography
2. support the extends for usr imgcat. maybe user modify the imgcat can use.

#### Possible Solutions
I will implement new imgcat by PIL resize images, show 2560*1918 images will be change to 500 * 374 show in the terminal.

**Comment by toonn** [comment]
I think it'd be fair to change the `ITerm2ImageDisplayer` class to resize images before viewing if this is really an issue with `imgcat`. Ideally though `imgcat` would be patched to be able to handle any size of image.

**Comment by Bugxyb** [comment]
yeah, but I think it not imgcat problem, the reason is image to large, b64 encoder string too long, send to image_decoder and become crash
this images is ok, and the b64 string length is 192600
![d29af8dce4291e4be630829a877ab49c](https://user-images.githubusercontent.com/6049660/75882428-1fb24380-5e5c-11ea-8373-fb2cbcc15be4.png)
but it cause image_decoder crash, and the b64 string length is 348620
![d333b6d13e92993596ef1c9b72bd0f13](https://user-images.githubusercontent.com/6049660/75882471-35c00400-5e5c-11ea-94d0-547358f5601a.png)

and maybe the Ranger Image preview will show the thumbnail better than original picture
